File: 0944_delete_columns_to_make_sorted/Solution.cs
// LeetCode 0944 - Delete Columns to Make Sorted
// https://leetcode.com/problems/delete-columns-to-make-sorted/

public class Solution {
    public int MinDeletionSize(string[] strs) {
        int ans = 0, m = strs[0].Length, n = strs.Length;
        for (int c = 0; c < m; c++) {
            for (int r = 0; r + 1 < n; r++) {
                if (strs[r][c] > strs[r + 1][c]) { ans++; break; }
            }
        }
        return ans;
    }
}

File: 0944_delete_columns_to_make_sorted/Solution.scala
// LeetCode 0944 - Delete Columns to Make Sorted
// https://leetcode.com/problems/delete-columns-to-make-sorted/

object Solution {
  def minDeletionSize(strs: Array[String]): Int = {
    var ans = 0
    val m = strs(0).length
    val n = strs.length
    var c = 0
    while (c < m) {
      var r = 0
      var bad = false
      while (r + 1 < n && !bad) {
        if (strs(r).charAt(c) > strs(r + 1).charAt(c)) { ans += 1; bad = true }
        r += 1
      }
      c += 1
    }
    ans
  }
}
